### ISSUE TITLE
Any chance to see the "LED indicator" for VideoDoorbell POE add to the entities?

### ISSUE THREAD

**Issue by Electronlibre2012** [JimStar/reolink_cctv#118]
Hello @JimStar 

in the Reolink APP i can toggle on/off the Led indicator who is arround the push button to call,

any chance to see this light entity in HA please?

Thanks

Olivier

**Comment by JimStar** [comment]
I don't have this model, so I can't analyze its traffic. You can do it yourself: run the admin UI of this camera, use the "switch on" LED, and see what command is sent to the camera. I need to know what is sent...  
You can use Chrome to enter the web-debug mode for Reolink camera, see the page 28 in their doc:  
[https://github.com/JimStar/reolink_ip/raw/master/doc/Camera%20HTTP%20API%20User%20Guide_v7.pdf](https://github.com/JimStar/reolink_ip/raw/master/doc/Camera%20HTTP%20API%20User%20Guide_v7.pdf).

**Comment by quizzical86** [comment]
Hi @JimStar 

This is what I get when I turn on the "Always on at night" setting for the LED light:
`getQueryDataByCount: flagO=[{"cmd":"SetIrLights","param":{"IrLights":{"channel":0,"state":"Auto"}},"action":0},{"cmd":"SetPowerLed","param":{"PowerLed":{"channel":0,"state":"On","eDoorbellLightState":"On"}},"action":0}]&countIdO=17&checkNumO=49&queryStringO=`

And when I turn it off:
`getQueryDataByCount: flagO=[{"cmd":"SetIrLights","param":{"IrLights":{"channel":0,"state":"Auto"}},"action":0},{"cmd":"SetPowerLed","param":{"PowerLed":{"channel":0,"state":"On","eDoorbellLightState":"Off"}},"action":0}]&countIdO=18&checkNumO=49&queryStringO=`

The Reolink app also supports quick replies. There are a couple of prerecorded messages (and you can also record your own messages) and you can select  that the doorbell should say that (e.g. I'm not home, leave the package at the door, i'll be there in a moment, etc). Thanks to your integration Home Assistant sends me a notification when someone's at the door together with a screenshot. It would be great if in that notification I could click on of those default messages. But in the browser I don't have the option to activate on of those quick replies. That's only in the Reolink app... Any way to get that working?

**Comment by JimStar** [comment]
Thanks, that debug output is useful. Will make use of it in the next version...

**Comment by Electronlibre2012** [comment]
thanks! sound good

**Comment by JimStar** [comment]
Doorbell light switch should now be available in [0.1.8](https://github.com/JimStar/reolink_cctv/releases/tag/0.1.8) (done some quick-fix as this new feature was broken in 0.1.7).  
Let me know if some issues with this new switch...

**Comment by Electronlibre2012** [comment]
thanks a lot! work like a charm! happy christmas and thanks again for your work on this integration!!!

**Comment by Electronlibre2012** [comment]
one info, perhaps it will help you to find out the NVR integration way : 

i plug my doorbell on a POE switch, all work great with the integration, i have now a Doorbell integration separate to my NVR integration.

I setup the channel 7 of my NVR to record 24/24/7 the doorbell and it is added to my NVR8-410, visitor sensor is there, no Led indicator.

The strange behavior is when the sensor visitor is on 4 minutes ago (of course i press on the doorbell button), the sensor_visitor_2 on the NVR integration become "anavailable" at the exact same time, after 1 minute the sensor state change to "off" on the NVR integration and "inactive" on the doorbell integration. Doorbell part work fine, no problem.

Its not a problem, it works, just perhaps one day, after a NVR update firwmare, we can have all together in the NVR integration, just more simple to have all together, but not a big deal...

have a good day

**Comment by rjmcfadd** [comment]
> Doorbell light switch should now be available in [0.1.8](https://github.com/JimStar/reolink_cctv/releases/tag/0.1.8) (done some quick-fix as this new feature was broken in 0.1.7). Let me know if some issues with this new switch...

After the 0.1.8 update, I now have a doorbell light switch for all of the cameras.
![image](https://user-images.githubusercontent.com/22861530/208172629-60c02bdb-9308-4c00-9b67-81aa04cafaed.png)

**Comment by JimStar** [comment]
Weird. Can't reproduce on my side... I need the full debug log (starting with HA restart) to see what's happening - see an issue-template for the config lines.

**Comment by JimStar** [comment]
Just in case: did you update to **0.1.7** before **0.1.8**? The 0.1.7 was available for a very short time before I deleted this version due to a bug in there...

**Comment by quizzical86** [comment]
I noticed the status of this light doesn't synchronise with changes made in the Reolink app. E.g. If I turn on the light using the Reolink app, then Home Assistant will still show the light as being off.
I assume this is a limitation of Reolink that we have to live with?

**Comment by JimStar** [comment]
The HA component polls the statuses of switches every 1 minute, to not overload a network by a bulk of poll-traffic. So if you changed some status in Reolink app, it should be synchronized after about 1 minute in HA...  
Unfortunately there are no push-notifications from Reolink devices about status changes.

**Comment by quizzical86** [comment]
Thanks for the very speedy and very clear reply! I tested and you are fully right.

**Comment by rjmcfadd** [comment]
> Just in case: did you update to **0.1.7** before **0.1.8**? The 0.1.7 was available for a very short time before I deleted this version due to a bug in there...

No, I updated from 0.1.6 to 0.1.8. I'm working on the debug log now.

**Comment by rjmcfadd** [comment]
> Weird. Can't reproduce on my side... I need the full debug log (starting with HA restart) to see what's happening - see an issue-template for the config lines.

[home-assistant_reolink_cctv_2022-12-19T19-48-54.360Z.log](https://github.com/JimStar/reolink_cctv/files/10262296/home-assistant_reolink_cctv_2022-12-19T19-48-54.360Z.log)

Here is what I have after a restart. Thank you!

**Comment by JimStar** [comment]
@rjmcfadd

This log does not look like the **full** log driven by the config-lines shown in the issue-templates of this project... I see debug enabled for **other** integrations (which I don't need in the log), but I don't see any debug lines of this component in there.

In addition:

>2022-12-19 14:12:11.357 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration reolink_dev which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant

Have you read the README of this project? Especially the second warning in the "IMPORTANT notes" section?

**Comment by rjmcfadd** [comment]
> @rjmcfadd
> 
> This log does not look like the **full** log driven by the config-lines shown in the issue-templates of this project... I see debug enabled for **other** integrations (which I don't need in the log), but I don't see any debug lines of this component in there.
> 
> In addition:
> 
> > 2022-12-19 14:12:11.357 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration reolink_dev which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
> 
> Have you read the README of this project? Especially the second warning in the "IMPORTANT notes" section?

I had already disabled Reolink_dev but it was still in HACS. I just removed it from HACS and captured more logs.

[home-assistant_2022-12-19T20-29-22.822Z.log](https://github.com/JimStar/reolink_cctv/files/10262565/home-assistant_2022-12-19T20-29-22.822Z.log)

**Comment by JimStar** [comment]
This one is a proper log. I see the way to fix this, will be OK in next version.

What I see there in addition:

This is the Reolink command sent by the component to your NVR:  
```
{'cmd': 'GetPowerLed', 'action': 0, 'param': {'channel': 1}}
```
It asks the NVR to return the status of power-led for the camera connected to the second channel of the NVR (the channel-index is zero-based, "1" is the second channel).  
This is the response from your NVR:  
```
   {
      "cmd" : "GetPowerLed",
      "code" : 0,
      "value" : {
         "PowerLed" : {
            "channel" : 0,
            "eDoorbellLightState" : "NotSupport",
            "state" : "On"
         }
      }
   },
```
I see there is a Reolink firmware **bug** here: no matter what channel this command is sent for, the Reolink firmware **always** responds with the `channel 0` in the response. While the **data itself** in the response is for the correct channel that was requested...

Anyway, discussed many times already here in tickets: Reolink has bugs in their firmware, so doorbell camera will still not work properly for you if connected through NVR. Even in the next release of this component where **this** little issue gets workarounded. You need to connect it directly to your home network, and add its own integration entry in HA.

**Comment by JimStar** [comment]
Should be fixed in [0.1.9](https://github.com/JimStar/reolink_cctv/releases/tag/0.1.9).
